src/Converter/ValueConverter.php

```diff
@@ -43,7 +43,7 @@ public function convert(AttributeInterface $attribute, $value, string $localeCod
                 if (!array_key_exists('amount', $value)) {
                     throw new \LogicException('Amount key not found');
                 }
-                $amount = (string) $value['amount'];
+                $amount = number_format((float) ($value['amount']));
```

Comment:
I'd format it using [Symfony's Intl component NumberFormatter](https://github.com/symfony/Intl/blob/5.4/NumberFormatter/NumberFormatter.php#L355).